ably/options.go

```diff
@@ -595,6 +599,19 @@ func (opts *clientOptions) idempotentRESTPublishing() bool {
 	return opts.IdempotentRESTPublishing
 }
 
+func (opts *clientOptions) hasActiveInternetConnection() bool {
+	res, err := opts.httpclient().Get(internetCheckUrl)
+	if err != nil {
+		return false
+	}
+	defer res.Body.Close()
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return false
+	}
+	return string(data) == internetCheckOk
```

Comment:
The spec says
"If the request succeeds and the text “yes” is included in the body"
So we want `bytes.Contains(data, []byte("yes"))`

Reply:
It's the same right. I mean it currently accurately checks for `yes` in the response. We have test for the same

Reply:
What will happen if someone changes https://internet-up.ably-realtime.com/is-the-internet-up.txt so that it does not return a newline at the end?

Reply:
Then the test will fail and as a part of the fix, we need to update the url

Reply:
But the spec says nothing about the body returned by this request including a newline

Reply:
That's why we are not checking response body in the test. We only check if internet is available or not. 
How do you suggest we process response without explicit check for the same

Reply:
Ideally server shouldn't respond with newline, but it's being introduced somehow.

Reply:
Anyways, in the future they can change response to just letter `y` or nil response with status code as 200. This needs another discussion

Reply:
If it's just the string contains we want to check, we can do that too. But not sure it will make any difference as such

Reply:
I think we should do precisely what the spec says. 
So if the server at the other end changes in a way which is conformant with the spec, our builds will not stop working.

Reply:
Okay, I will update the code as suggested. Let me know if there are any other changes.

Reply:
updated

Reply:
Thanks